pallets/order-book/src/lib.rs

```diff
@@ -651,8 +644,8 @@ pub mod pallet {
         pub fn place_limit_order(
             origin: OriginFor<T>,
             order_book_id: OrderBookId<AssetIdOf<T>, T::DEXId>,
-            price: OrderPrice,
-            amount: OrderVolume,
+            price: Balance,
+            amount: Balance,
```

Comment:
We keep `Balance` only for extrinsics and some traits (like `LiquiditySource`). In all other places we want to use `BalanceUnit`.
It could be pretty good to use `BalanceUnit` instead of `Balance` in all Sora pallets, but it is outscope of my PR))
Yes, some `Balance` fields could be interpreted as divisible or indivisible, but it is how it already works besides my solution. The problem was introduced at the beginning when it was decided to use different precisions inside the network and have 0 precision for NFT. We cannot fix it right now.